src/cinder/audio/cocoa/DeviceManagerAudioSession.mm

```diff
@@ -264,18 +264,21 @@ @implementation AudioSessionNotificationHandlerImpl
 
 - (void)handleInterruption:(NSNotification *)notification
 {
-	NSUInteger interruptionType = [[notification.userInfo valueForKey:AVAudioSessionInterruptionTypeKey] unsignedIntValue];
+	NSUInteger interruptionType = [[notification.userInfo objectForKey:AVAudioSessionInterruptionTypeKey] unsignedIntValue];
 
 	if( interruptionType == AVAudioSessionInterruptionTypeBegan ) {
 		mDeviceManager->beginInterruption();
 	}
 	else if( interruptionType == AVAudioSessionInterruptionTypeEnded ) {
-		NSUInteger interruptionOption = [[notification.userInfo valueForKey:AVAudioSessionInterruptionOptionKey] unsignedIntValue];
-		if( interruptionOption == AVAudioSessionInterruptionOptionShouldResume ) {
-			mDeviceManager->endInterruption(true);
-		} else {
-			mDeviceManager->endInterruption(false);
-		}
+		UIApplication *app = [UIApplication sharedApplication];
+		AVAudioSession *session = [AVAudioSession sharedInstance];
+
+		auto resumeImmediately = app.applicationState == UIApplicationStateActive ||
```

Comment:
Hmm.. Well I assume we would still need to check if the UIApplication is active when we connect to `signalDidBecomeActive`, and keep track of that variable ourselves.. Seems simpler to just get it through the `sharedInstance` since we have to assume that exists anyway.. Of course if it's guaranteed that an application will be active when the DeviceManager is created we wouldn't need to check.. But since it's lazy loaded, we don't know.